### PR TITLE
Add jest tests for `create-toolpad-app`

### PR DESCRIPTION
**Pull request by bharatkashyap**
- Closes https://github.com/mui/mui-toolpad/issues/1901

### CODE REVIEW

File: packages/create-toolpad-app/jest.config.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { Config } from 'jest';
+
+export default {
+  preset: 'ts-jest',
+  transform: {},
```
Comment (Janpot) [inline packages/create-toolpad-app/jest.config.ts:5]:
```suggestion
```
Is this needed?

**Comment by Janpot** [comment]
Have you seen the test output? https://app.circleci.com/pipelines/github/mui/mui-toolpad/7236/workflows/992b6458-36d0-496e-b685-e7297008af48/jobs/31467/parallel-runs/0/steps/0-113

To avoid async issues, avoid using callbacks and `done`. Best to use promises all the way. An async function that works with callbacks can easily be converted to a promise returning one

File: packages/create-toolpad-app/src/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
+const { spawn, exec } = require('child_process');
```
Comment (Janpot) [inline packages/create-toolpad-app/src/index.spec.ts:1]: avoid commonjs

File: packages/create-toolpad-app/src/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            done();
+          }
+        });
+      }
+    });
+  }, 60000);
+
+  it('should return a 200 status code when hitting the health-check route', (done) => {
+    http.get(`${toolpadAppAddress}/health-check`, (res: ServerResponse) => {
+      expect(res.statusCode).toBe(200);
```
Comment (Janpot) [inline packages/create-toolpad-app/src/index.spec.ts:58]: This won't be caught by the test. Best to use promises

**Comment by Janpot** [comment]
I've pushed a commit proposing an updated setup that may be easier to work with

File: packages/create-toolpad-app/tests/index.spec.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        '@mui/toolpad': expect.any(String),
+      }),
+      scripts: expect.objectContaining({
+        build: 'toolpad build',
+        dev: 'toolpad dev',
+        start: 'toolpad start',
+      }),
+    }),
+  );
+});
```
Comment (Janpot) [inline packages/create-toolpad-app/tests/index.spec.ts:39]: Weren't you going to try running the app in this test as well?

Comment (bharatkashyap) [reply packages/create-toolpad-app/tests/index.spec.ts:39]: Yes, I'm currently on that (also adding the `no-install` option to make running this test quicker)

**Review by Janpot — APPROVED** [review]

**Pull request merged**